bc_obps/compliance/service/supplementary_version_service.py

```diff
@@ -39,6 +38,98 @@ def handle(
+            # Return True if the previous version has an obligation with no invoice
+            print('before no invoice check')
+            return SupplementaryVersionService._obligation_has_no_invoice(previous_compliance_report_version)
+        if previous_summary.credited_emissions > ZERO_DECIMAL:
+            # Return True if the previous version has earned credits that have not been requested
+            return SupplementaryVersionService._earned_credits_not_issued(previous_compliance_report_version)
+        return False
+
+    @staticmethod
+    def handle(
+        compliance_report: ComplianceReport,
+        new_summary: ReportComplianceSummary,
+        previous_summary: ReportComplianceSummary,
+        version_count: int,
+    ) -> Optional[ComplianceReportVersion]:
+
+        previous_compliance_version = (
+            SupplementaryVersionService._get_previous_compliance_version_by_report_and_summary(
+                compliance_report, previous_summary
+            )
+        )
+        # Update previous version status to SUPERCEDED
+        previous_compliance_version.status = ComplianceReportVersion.ComplianceStatus.SUPERCEDED
+        previous_compliance_version.save()
+
+        # Create new version
+        compliance_report_version = ComplianceReportVersion.objects.create(
+            compliance_report=compliance_report,
+            report_compliance_summary=new_summary,
+            status=ComplianceReportVersion.ComplianceStatus.NO_OBLIGATION_OR_EARNED_CREDITS,
+            is_supplementary=True,
+            previous_version=previous_compliance_version,
+        )
+        print('before delete')
+        # Handle supercede obligation
+        if previous_summary.excess_emissions > ZERO_DECIMAL:
+            # Delete hanging superceded obligation record
+            ComplianceObligation.objects.get(compliance_report_version=previous_compliance_version).delete()
+            print('after delete')
+
+        # Handle supercede earned credit
+        if previous_summary.credited_emissions > ZERO_DECIMAL:
+            # Delete hanging superceded earned credit record
+            ComplianceEarnedCredit.objects.get(compliance_report_version=previous_compliance_version).delete()
+
+        # Create new obligation record if new version has excess emissions
+        if new_summary.excess_emissions > ZERO_DECIMAL:
+            # Create new obligation
+            obligation = ComplianceObligationService.create_compliance_obligation(
+                compliance_report_version.id, new_summary.excess_emissions
+            )
+            ElicensingObligationService.handle_obligation_integration(
+                obligation.id, compliance_report.compliance_period
+            )
+            compliance_report_version.status = (
+                ComplianceReportVersion.ComplianceStatus.OBLIGATION_PENDING_INVOICE_CREATION
+            )
+            compliance_report_version.save()
```

Comment:
Do we need this part? We have similar code in `handle_obligation_integration` function